comment_scan.rb

```diff
@@ -85,24 +85,23 @@ def restart(num_to_post, bundle)
             end
           when 'huh?'
             matched_regexes = report_raw(comment["post_type"], comment["body_markdown"])
-            reason_text = ""
-            #Go through regexes we matched to build reason_text
-            matched_regexes.each { |regex_matched| reason_text += "Matched reason \"#{Reason.where(id: regex_matched["reason_id"])[0]["name"]}\" for regex: #{regex_matched["regex"]}\n" }
-            #If post isn't deleted, check if this was an inactive comment
-            if !isPostDeleted(cli, comment["post_id"])
-              post = Array(cli.posts(comment["post_id"].to_i))[0]
-              if Time.at(post.json["last_activity_date"].to_i).to_date < Time.at(comment["creation_date"].to_i).to_date - 30
-                reason_text += "Comment was made #{(Time.at(comment["creation_date"].to_i).to_date - Time.at(post.json["last_activity_date"].to_i).to_date).to_i} days after last activity on post\n"
+            # Go through regexes we matched to build reason_text
+            reason_text = matched_regexes.map { |rm| "Matched reason \"#{Reason.where(id: rm["reason_id"]).first.name}\" for regex: #{rm.regex}" }.join("\n")
+            # If post isn't deleted, check if this was an inactive comment
+            if !post_deleted?(cli, comment["post_id"])
```

Comment:
Now that I'm looking at the `post_deleted?` method, there's a pretty cool ruby thing you can do. Currently, you use one API call to check if it's deleted, and another (identical) API call to get the comments on the post. A quick diversion into some ruby awesomeness: Every value can be "truthy" or "falsey". This is super handy for situations like this. Here, you could make `post_deleted?` return either `nil` or an array, and make this `if` statement `if post = post_deleted?`. This would work because if post_deleted? returns nil, the expression will be falsey and so the code inside the if won't execute, and if post_deleted? returns an array, which is truthy, the expression with by truthy and the code inside will be executed with `post` already set to the correct post.

This would also allow you to merge the two if statements.